odoo/addons/base/models/ir_actions.py

```diff
@@ -573,9 +621,11 @@ def _default_update_path(self):
     ir_cron_ids = fields.One2many('ir.cron', 'ir_actions_server_id', 'Scheduled Action', context={'active_test': False})
     # Python code
     code = fields.Text(string='Python Code', groups='base.group_system',
-                       default=DEFAULT_PYTHON_CODE,
                        help="Write Python code that the action will execute. Some variables are "
                             "available for use; help about python expression is given in the help tab.")
+    code_history = fields.Json(string='Python Code History', copy=False)
```

Comment:
Ho great! A new nail! Why storing this as a json field instead of having a model `ir.act.server.history`?

Reply:
No json please. We have a relational database.